plugins/guests/freebsd/cap/mount_nfs_folder.rb

```diff
@@ -1,22 +1,41 @@
+require "vagrant/util/retryable"
+
 module VagrantPlugins
   module GuestFreeBSD
     module Cap
       class MountNFSFolder
+        extend Vagrant::Util::Retryable
+
         def self.mount_nfs_folder(machine, ip, folders)
           comm = machine.communicate
 
           commands = []
 
-          folders.each do |_, opts|
-            if opts[:nfs_version]
-              mount_opts = "-o nfsv#{opts[:nfs_version]}"
+          folders.each do |name, opts|
+            # Expand the guest path so we can handle things like "~/vagrant"
+            expanded_guest_path = machine.guest.capability(
+              :shell_expand_guest_path, opts[:guestpath])
+
+            # Do the actual creating and mounting
+            commands << "mkdir -p '#{expanded_guest_path}'"
+
+            # Mount
+            hostpath = opts[:hostpath].dup
+            hostpath.gsub!("'", "'\\\\''")
+
+            # Figure out any options
+            mount_opts = ["nfsv#{opts[:nfs_version]}"]
+            mount_opts << "udp" if opts[:nfs_udp]
+            if opts[:mount_options]
+              mount_opts = opts[:mount_options].dup
```

Comment:
Correct, I just copied https://github.com/mitchellh/vagrant/blob/master/plugins/guests/linux/cap/mount_nfs.rb and modified some things.

Want me to submit a fix for Linux too?